common/structs.go

```diff
@@ -0,0 +1,97 @@
+// Structs contain all the structures used by the Infra CLI, Server and Client
+// Eache struct has a barshal() and an unmarshal() method to convert them to JSON and back
+// They are all the implementation of the JSONable interface, which requires these two methods.
+
+package common
+
+import (
+	"encoding/json"
+)
+
+var (
+	// AdminPass is the password used to authenticate the admin.
+	AdminPass string
+	// APIURL is the address of the infra-server's api. By default it will be "http://localhost:7474/api".
+	APIURL string
+)
+
+// JSONable all struct which implements the marshal and unmarshal methods, to convert it to JSON and back.
+type JSONable interface {
```

Comment:
There's a built'n interface for both serialisation and deserialisation which would be implemented by structs those require custom serialisation logic for any reason.

http://choly.ca/post/go-json-marshalling/